### PR TITLE
Fix minor issues

### PR DESCRIPTION
**Pull request by haslinghuis**
Minor issues found and fixed:

- [x] `connectionType` not established before connecting.
- [x] Removed unused variable `offset`
- [x] Moved logging of `crc` errors out of `msp.js` to better show where it's coming from
- [x] Intermittent undefined `code` in callback after save and reboot
```
Error in event handler: TypeError: Cannot read property 'code' of undefined
    at MspHelper.process_data (chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/msp/MSPHelper.js:1624:38)
    at chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/msp.js:226:13
    at Array.forEach (<anonymous>)
    at Object.notify (chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/msp.js:225:24)
    at Object._dispatch_message (chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/msp.js:220:14)
    at Object.read (chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/msp.js:182:22)
    at read_serial (chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/js/serial_backend.js:562:13)
```
- [x] Intermittent beginning to see `Reboot request accepted` by changing timeout from `100` to `200` in `serial_backend.js`
```
MSPHelper.js:632 Settings Saved in EEPROM
MSPHelper.js:723 Reboot request accepted
serial.js:404 serial: receive error: UNDEFINED: device_lost
serial.js:217 serial: closed connection with ID: 4, Sent: 7077 bytes, Received: 79082 bytes
```

### CODE REVIEW

File: src/js/msp/MSPHelper.js [hunk 1]
```diff
@@ -1620,20 +1620,22 @@ MspHelper.prototype.process_data = function(dataHandler) {
...
-            clearInterval(dataHandler.callbacks[i].timer);
-
-            // remove object from array
-            dataHandler.callbacks.splice(i, 1);
-            if (!crcError || callbackOnError) {
-                // fire callback
-                if (callback) callback({'command': code, 'data': data, 'length': data.byteLength, 'crcError': crcError});
+    for (let i = dataHandler.callbacks.length - 1; i >= 0; i--) { // iterating in reverse because we use .splice which modifies array length
+        if (dataHandler.callbacks[i] !== undefined) {
+            if (dataHandler.callbacks[i].code === code) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:1625]: More concise to go `if (dataHandler.callbacks[i]?.code === code) {`.

**Review by mikeller — COMMENTED** [review]
Your rummaging around to fix the TCP connection is finding some valuable fixes. Thanks!

File: src/js/msp/MSPHelper.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/msp/MSPHelper.js:1625]: For reference: https://github.com/claudepache/es-optional-chaining

**Comment by haslinghuis** [comment]
It's recommended to replace all `var` variables to `let` or `const`.
I can do it here or in a separate PR per file.

File: src/js/msp/MSPHelper.js [hunk 2]
```diff
@@ -1620,20 +1619,22 @@ MspHelper.prototype.process_data = function(dataHandler) {
...
-            clearInterval(dataHandler.callbacks[i].timer);
-
-            // remove object from array
-            dataHandler.callbacks.splice(i, 1);
-            if (!crcError || callbackOnError) {
-                // fire callback
-                if (callback) callback({'command': code, 'data': data, 'length': data.byteLength, 'crcError': crcError});
+    for (let i = dataHandler.callbacks.length - 1; i >= 0; i--) { // iterating in reverse because we use .splice which modifies array length
+        if (dataHandler.callbacks[i] !== undefined) {
+            if (dataHandler.callbacks[i]?.code === code) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:1624]: In this case (`?.`) the outer `if (dataHandler.callbacks[i] !== undefined) {` is not needed anymore, as `dataHandler.callbacks[i]?.code` will yield `undefined` if `dataHandler.callbacks[i]` does not exist. :-)

Comment (haslinghuis) [reply src/js/msp/MSPHelper.js:1624]: Err, one Control-Z to many. Fixed

**Review by mikeller — APPROVED** [review]

**Pull request merged**